docs/en/02_Developer_Guides/08_Performance/02_HTTP_Cache_Headers.md

```diff
@@ -1,7 +1,125 @@
+Getting it wrong can accidentally expose draft pages or other protected content.
+The [Google Web Fundamentals](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching#public_vs_private)
+are a great way to learn about HTTP caching.
+
+## Cache Control Headers
+
+In order to support developers in making safe choices around HTTP caching,
+we're using a `HTTPCacheControl` class to control if a response
+should be considered public or private. This is an abstraction on existing
+lowlevel APIs like `HTTP::add_cache_headers()` and `SS_HTTPResponse->addHeader()`.
+
+The `HTTPCacheControl` API makes it easier to express your caching preferences
+without running the risk of overriding essential core safety measures.
+Most commonly, these APIs will prevent HTTP caching of draft content.
+
+It will also prevent content generated with an active session,
+since the system can't tell whether session data was used to vary the output.
+In this case, it's up to the developer to opt-in to caching,
+after ensuring that certain execution paths are safe despite of using sessions.
+
+The system behaviour does not guard against accidentally caching "private" content,
+since there are too many variations under which output could be considered private
+(e.g. a custom "approval" flag on a comment object). It is up to
+the developer to ensure caching is used appropriately there.
+
+The `HTTPCacheControl` class supplements the `HTTP` helper class.
+It comes with methods which let developers safely interact with the `Cache-Control` header.
+
+ * `disableCache()`: Disables caching in intermediary caches as well as HTTP clients.
+    Removes all state and replaces it with `no-cache, no-store, must-revalidate`.
+ * `privateCache()`: Disables caching in intermediary caches, but allows it in HTTP clients.
+    Sends `Cache-Control: private`.
+ * `publicCache()`: Enables caching in intermediary caches and HTTP clients.
+    Sends `Cache-Control: public`.
+    
+Each of these highlevel methods has a boolean `$force` parameter which determines
+their application priority regardless of execution order.
+The priority order is as followed, sorted in descending order
+(earlier items will overrule later items): 
+
+ * `disableCache($force=true)`
+ * `privateCache($force=true)`
+ * `publicCache($force=true)`
+ * `disableCache()`
+ * `privateCache()`
+ * `publicCache()`
+
+## Cache Control Examples
+
+Globally opting into HTTP caching for all page content.
+SilverStripe will still override this preference when a session is active,
+or draft content has been requested. 
+
+```php
+class Page_Controller extends ContentController
+{
+    public function init()
+    {
+        HTTPCacheControl::inst()
+           ->publicCache()
+           ->setMaxAge(60); // 1 minute
+
+        
+        parent::init();
+    }
+}
+```
+
+Opting out of HTTP caching in a particular controller action.
```

Comment:
This example only opts out of caching in CDNs/proxies, we may want to make that clearer or change the example

Reply:
I'll change it to `disableCache()` since that's more likely the preferred action